### PR TITLE
[WIP]added benchmarks/symbench.m

### PR DESCRIPTION
**Pull request by abhinavagarwal07**
@isuruf 
Does this look ok?

### CODE REVIEW

**Comment by isuruf** [comment]
For garbage collected languages like Mathematica, you need to clear everything before a benchmark.

**Comment by isuruf** [comment]
Btw, add `[skip ci]` to the front of each message where you don't want travis and appveyor to run

**Comment by abhinavagarwal07** [comment]
`symbench.cpp` doesn't take any command line argument.
How will we plot the graph in this case ?

**Comment by isuruf** [comment]
This should just be a table I guess

**Comment by abhinavagarwal07** [comment]
@isuruf ping

File: benchmarks/symbench.m [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+(*define R3*)
+f[]=x+y+z;
+Array[fc, 10];
+"Time for R3: " 
+Print[AbsoluteTiming[Do[fc[i] =Boole[(f[]==f[])],{i,1,10}]]*1000]
+
+(*define R7*)
+"Time for R7: " 
+f[x_]=x^24+34* x^12+45*x^3+9* x^18+34* x^10+32*x^21;
```
Comment (isuruf) [inline benchmarks/symbench.m]: This does not involve symbolics. What you should do is to create the expression first and then measure the time taken to replace `x` with `0.5`

File: benchmarks/symbench.m [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+R2[n_,y_]:=hermite[n,y];
+n=15;
+"Time for R2 : " 
+Print[1000*AbsoluteTiming[R2[n,y];]];
+Remove["Global`*"]
+ClearAll["Global`*"]
+
+(*define R3*)
+f[]=x+y+z;
+Array[fc, 10];
```
Comment (isuruf) [inline benchmarks/symbench.m]: What does this do?

**Comment by isuruf** [comment]
By the way, `benchmarks/expand2.nb` has a notebook implementing most of the benchmarks, can you take a look?

File: benchmarks/symbench.m (hunk 2 above)
Comment (abhinavagarwal07) [inline benchmarks/symbench.m]: Creates an array of size 10.
using `Table` can be an alternative.

**Comment by abhinavagarwal07** [comment]
@isuruf 
Didn't find `expand2.nb` where is it? 

What can be used analogous to `std::set` in mathematica ?

**Comment by isuruf** [comment]
I meant symbench.nb, https://github.com/symengine/symengine/blob/master/benchmarks/symbench.nb

**Comment by abhinavagarwal07** [comment]
In `benchmarks/symbench.cpp`, R3()
```c++
    std::vector<bool> vec(10);
    auto t1 = std::chrono::high_resolution_clock::now();
    for (int i = 0; i < 10; i++) {
        vec.push_back(eq(*f, *f));
    }
```
I think instead of `std::vector<bool> vec(10);` -> `vec.reserve(10);` was intended

`R5()`
`.cpp` uses `set` while `.nb` uses `list`. So I don't think it would be a fair competition.

`R7()`
`.cpp` substitutes value `0.5` while `.nb` substitues random number.

otherwise `symbench.nb` looks perfect.

**Comment by isuruf** [comment]
Agree about `R3` and `R7`. `R5` is okay, because that's how Mathematica represents sets.

**Comment by abhinavagarwal07** [comment]
#977

**Pull request closed without merging**